### PR TITLE
Use category page and sidebar for libraries overview

### PR DESCRIPTION
**Pull request by jlvandenhout**
# Description of change

This replaces the libraries section of the Wiki with a category page showing an overview of the libraries and sidebar with direct links to the libraries.

## Links to any relevant issues

closes issue #630

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Local Wiki build

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested the wiki locally and tested that all external links work

### CODE REVIEW

**Review by Dr-Electron — APPROVED** [review]

File: src/components/HomeLayout/LibrariesSection.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      />
+      <div className='libraries__section'>
+        <div className='libraries__head'>
+          <div className='libraries__icon'>
+            <Identity />
+          </div>
+          <Languages
+            languages={{
+              Rust: 'identity.rs/libraries/rust/getting_started',
+              WASM: 'identity.rs/libraries/wasm/getting_started',
```
Comment (lucas-tortora) [inline src/components/HomeLayout/LibrariesSection.tsx:50]:
```suggestion
              Rust: '/identity.rs/libraries/rust/getting_started',
              WASM: '/identity.rs/libraries/wasm/getting_started',
```

File: src/components/HomeLayout/LibrariesSection.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+          <li className='libraries__feature'>
+            Verifiable Credentials (VC)
+          </li>
+        </ul>
+        <Link
+          to='identity.rs/introduction'
+          className='libraries__button button button--outline button--primary'
+        >
+          Create an identity
+        </Link>
```
Comment (lucas-tortora) [inline src/components/HomeLayout/LibrariesSection.tsx:69]:
```suggestion
        <ul className='libraries__features'>
          <li className='libraries__feature'><a href={"'/identity.rs/getting_started/create_and_publish"}>Identity framework</a></li>
          <li className='libraries__feature'>
            <a href={"/identity.rs/concepts/decentralized_identifiers/overview"}>Decentralized Identifiers (DID)</a>
          </li>
          <li className='libraries__feature'>
            <a href={"/identity.rs/concepts/verifiable_credentials/overview"}>Verifiable Credentials (VC)</a>
          </li>
        </ul>
        <Link
          to='/identity.rs/getting_started/create_and_publish'
          className='libraries__button button button--outline button--primary'
        >
          Create an identity
        </Link>
```

File: src/components/HomeLayout/LibrariesSection.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            Create messages and transactions
+          </li>
+          <li className='libraries__feature'>Sign transactions</li>
+        </ul>
+        <Link
+          to='iota.rs/welcome'
+          className='libraries__button button button--outline button--primary'
+        >
+          Interact with the network
+        </Link>
```
Comment (lucas-tortora) [inline src/components/HomeLayout/LibrariesSection.tsx:104]:
```suggestion
        <ul className='libraries__features'>
          <li className='libraries__feature'>
            <a href={"/iota.rs/examples/get_info"}> Interact with the IOTA network</a>
          </li>
          <li className='libraries__feature'>
            <a href={"/iota.rs/examples/simple_message"}>Create messages</a> and <a href={"iota.rs/examples/transaction"}>transactions</a>
          </li>
          <li className='libraries__feature'>
            <a href={"/iota.rs/examples/data_message"}>
              Send data messages
            </a>
          </li>
        </ul>
        <Link
          to='/iota.rs/welcome'
          className='libraries__button button button--outline button--primary'
        >
          Interact with the network
        </Link>
```

**Review by lucas-tortora — CHANGES_REQUESTED** [review]
* Please add `/` before links, if not the redirect to `/categories/`. 
* The features you list in each card were originally calls to action.  We've got a notion doc with a couple for each lib, please let me know if you need a link.  I've added a few examples in the comments.  I can create a PR for the home page if you want, just let me know. 
* TBH, I don't see much send in removing the section and replacing it with the home page content.  If we want to go in that direction, I'd make the navbar direct to the `libraries` header on the home page using a # link.  WDYT?

**Comment by jlvandenhout** [comment]
> * Please add `/` before links, if not the redirect to `/categories/`. 

Thanks, I missed those.

> * The features you list in each card were originally calls to action.  We've got a notion doc with a couple for each lib, please let me know if you need a link.  I've added a few examples in the comments.  I can create a PR for the home page if you want, just let me know. 

The component you changed is used on both the landing page and the overview page, so this will change both already.

> * TBH, I don't see much send in removing the section and replacing it with the home page content.  If we want to go in that direction, I'd make the navbar direct to the `libraries` header on the home page using a # link.  WDYT?

We discussed and A/B tested this in the Wiki team and having the separate overview page felt more natural. Also this allows the landing page to remain a feature advertisement and the overview pages complete references. This PR is just an initial replacement of the current overview page.

What we actually have in mind is provide a better introduction text and use the Docusaurus category item to automate the card creation, based on sidebar links. But we are waiting for facebook/docusaurus#7818 to provide us with a means to set some metadata on the links to auto-generated cards that are more close to the cards we had before, with the icon and description.

File: src/components/HomeLayout/LibrariesSection.tsx (hunk 2 above)
Comment (jlvandenhout) [reply src/components/HomeLayout/LibrariesSection.tsx]: I left the CTAs as plain text for now, if still applicable we can open a new issue and PR.

**Comment by lucas-tortora** [comment]
> * The features you list in each card were originally calls to action.  We've got a notion doc with a couple for each lib, please let me know if you need a link.  I've added a few examples in the comments.  I can create a PR for the home page if you want, just let me know.

It'd be nice if you could add the links in the calls to action.  Each of the phrases in the cards were requested by Annie to try and drive users to the docs using CTAs like `Connect to the network`, `Send a transaction`, etc. 

> * TBH, I don't see much send in removing the section and replacing it with the home page content.  If we want to go in that direction, I'd make the navbar direct to the `libraries` header on the home page using a # link.  WDYT?

The content matching isn't really the point here, the point is that it's duplicate content,     the current overview has content which is better suited as such.

File: src/components/HomeLayout/LibrariesSection.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            <IotaCore />
+          </div>
+          <Languages
+            languages={{
+              Rust: '/iota.rs/getting_started/rust',
+              NodeJS: '/iota.rs/getting_started/nodejs',
+              Python: '/iota.rs/getting_started/python',
+              Java: '/iota.rs/getting_started/java/getting_started',
+              Wasm: '/iota.rs/getting_started/wasm',
+              Javascript: '/iotajs/welcome',
```
Comment (lucas-tortora) [inline src/components/HomeLayout/LibrariesSection.tsx:80]:
```suggestion
              Typescript: '/iotajs/welcome',
```

**Review by lucas-tortora — APPROVED** [review]
Just a label

**Comment by jlvandenhout** [comment]
> It'd be nice if you could add the links in the calls to action.  Each of the phrases in the cards were requested by Annie to try and drive users to the docs using CTAs like `Connect to the network`, `Send a transaction`, etc. 

Good point, willdo!

**Comment by jlvandenhout** [comment]
@lucas-tortora @Dr-Electron I updated at least the library CTAs, could you have a look again? Later we can update the CTAs for nodes and think about a better overview page for the libraries, but I guess this will do for now?

**Review by Dr-Electron — APPROVED** [review]

File: .gitmodules [hunk 5]
```diff
@@ -94,3 +94,15 @@
...
+	url = https://github.com/iotaledger/inx-mqtt.git
+	branch = develop
+[submodule "content/build/wasp/production"]
+	path = content/build/wasp/production
+	url = https://github.com/iotaledger/wasp
+	branch = master
+[submodule "content/build/inx-poi/develop"]
+	path = content/build/inx-poi/develop
+	url = https://github.com/iotaledger/inx-poi
+  branch = develop
```
Comment (Dr-Electron) [inline .gitmodules:108]:
```suggestion
	url = https://github.com/iotaledger/inx-poi
        branch = develop
```

**Pull request merged**